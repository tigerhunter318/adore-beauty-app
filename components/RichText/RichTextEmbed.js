import React from 'react'
import { StyleSheet, View } from 'react-native'
import HTMLView from 'react-native-htmlview'
import DOMSerializer from 'dom-serializer'
import Container from '../ui/Container'
import EmbedWebView from '../ui/EmbedWebView'
import Type from '../ui/Type'
import Icon from '../ui/Icon'
import theme from '../../constants/theme'
import { vw } from '../../utils/dimensions'
import { getIn } from '../../utils/getIn'
import PodcastRichTextWidget from '../podcasts/PodcastRichTextWidget'
import CustomWebView from '../ui/CustomWebView'
import { sanitizeContent } from '../../utils/format'

const htmlStyles = StyleSheet.create({
  p: {
    fontSize: 16,
    lineHeight: 24,
    color: theme.lightBlack,
    letterSpacing: 0.1
  },
  h2: {
    fontSize: 18,
    lineHeight: 26,
    fontWeight: '700',
    marginTop: 10,
    marginBottom: 10
  },
  h3: {
    fontSize: 15,
    lineHeight: 24,
    fontWeight: '700',
    color: theme.lightBlack
  },
  strong: {
    fontWeight: '700'
  },
  a: {
    flex: 1,
    textDecorationLine: 'underline'
  }
})

const styles = StyleSheet.create({
  videoContainer: {
    flex: 1,
    alignContent: 'center'
  },
  WebViewStyle: {
    height: 400,
    width: 300,
    alignSelf: 'center',
    alignContent: 'center',
    margin: 20
  },
  tiktokWebViewStyle: {
    height: 583,
    width: 'auto',
    marginTop: 40
  },
  blockquote: {
    letterSpacing: 1.5,
    lineHeight: 28
  }
})

const RichTextEmbedVideo = ({ node }) => {
  const videoURL = node?.children?.[0]?.attribs?.src

  if (!videoURL) return null

  return (
    <View style={styles.videoContainer}>
      <CustomWebView
        source={{
          html: `
        <video width="100%" height="100%" controls style="background-color: ${theme.black}">
          <source src="${videoURL}" type="video/mp4">
        </video>
        `
        }}
        style={styles.WebViewStyle}
      />
    </View>
  )
}

const RichTextEmbedTikTokVideo = ({ node }) => {
  const uri = getIn(node, 'attribs.cite')

  if (!uri) return null

  return (
    <Container mb={3}>
      <CustomWebView
        scrollEnabled={false}
        source={{ uri }}
        style={styles.tiktokWebViewStyle}
        mediaPlaybackRequiresUserAction
      />
    </Container>
  )
}

const RichTextEmbedBlockQuote = ({ node }) => (
  <Container>
    {node?.children?.map((item, keyIndex) => {
      if (!item?.children?.[0]?.data?.trim()) return null

      return (
        <Type bold size={23} center heading key={`blockquote-content-${keyIndex}`} style={styles.blockquote}>
          {item.children[0].data}
        </Type>
      )
    })}
  </Container>
)

export const RichTextEmbedWebview = ({ node, height }) => {
  const w = getIn(node, 'attribs.width')
  const h = getIn(node, 'attribs.height')
  let html = DOMSerializer(node)
  let viewHeight

  if (getIn(node, 'attribs.class') === 'instagram-media') {
    const instagramNode = {
      ...node,
      attribs: {
        ...node.attribs,
        style: `${node.attribs?.style}; max-width:100% !important; min-width:auto !important;`
      }
    }
    html = DOMSerializer(instagramNode)
  }
  if (w) {
    html = html.replace(w, '100%')
  }
  if (h) {
    viewHeight = parseInt(h.toString())
    const ratio = (vw() - 20) / w
    viewHeight = parseInt(viewHeight * ratio)
    html = html.replace(h, viewHeight)
  }

  return (
    <Container mb={2}>
      <EmbedWebView html={html} height={height || viewHeight} />
    </Container>
  )
}
const RichTextEmbedOrderedListItem = ({ item, keyIndex }) => (
  <Container ph={2} rows mt={1.5}>
    <Type size={16} style={{ marginRight: 12, textAlign: 'right', width: 22 }}>
      {keyIndex + 1}.
    </Type>
    <Type size={16}>{item.children[0].children[0].data}</Type>
  </Container>
)
const RichTextEmbedListItem = ({ item }) => (
  <Container ph={2} rows mt={1.5}>
    <Icon type="adore" name="check-item" size={18} color={theme.darkGray} style={{ marginRight: 12 }} />
    <Type size={16}>{item.children[0].children[0].data}</Type>
  </Container>
)

const NodeComponent = ({ node, nodeIndex }) => {
  const { name, children, attribs } = node

  if (name === 'p') {
    if (children?.length > 0 && children?.[0]?.name === 'video') {
      return <RichTextEmbedVideo node={node} />
    }
  }
  if (name === 'ul') {
    return children?.map((item, keyIndex) => <RichTextEmbedListItem item={item} key={`li-${nodeIndex}-${keyIndex}`} />)
  }
  if (name === 'ol') {
    return children?.map((item, keyIndex) => (
      <RichTextEmbedOrderedListItem item={item} keyIndex={keyIndex} key={`ol-${nodeIndex}-${keyIndex}`} />
    ))
  }

  if (name === 'blockquote') {
    if (attribs?.class === 'instagram-media') {
      return <RichTextEmbedWebview node={node} />
    }

    if (attribs?.class === 'tiktok-embed') {
      return <RichTextEmbedTikTokVideo node={node} />
    }

    return <RichTextEmbedBlockQuote node={node} />
  }

  if (name === 'a') {
    if (attribs?.class === 'embedly-card') {
      let height

      if (attribs?.href?.includes('omny.fm')) {
        height = 220
        return <PodcastRichTextWidget url={attribs.href} />
      }
      if (attribs?.href?.includes('media.giphy.com')) {
        height = 300
      } else {
        height = 500
      }
      return <RichTextEmbedWebview node={node} height={height} />
    }
  }

  if (name === 'iframe' || children?.[0]?.name === 'iframe') {
    if (attribs?.src?.includes('omny.fm')) {
      return (
        <Container pt={2}>
          <PodcastRichTextWidget url={attribs.src} />
        </Container>
      )
    }

    return <RichTextEmbedWebview node={node} />
  }

  return null
}

const RichTextEmbed = ({ content }) => {
  const renderNode = (node, index, siblings, parent, defaultRenderer) => {
    try {
      return <NodeComponent node={node} key={`${node && node.name}-${index}`} nodeIndex={index} />
    } catch (error) {
      console.warn('RichTextEmbed', 'renderNode', error)
    }
  }

  return (
    <Container>
      <HTMLView
        value={sanitizeContent(content.externalEmbed)}
        addLineBreaks={false}
        stylesheet={htmlStyles}
        renderNode={renderNode}
      />
    </Container>
  )
}

export default RichTextEmbed
