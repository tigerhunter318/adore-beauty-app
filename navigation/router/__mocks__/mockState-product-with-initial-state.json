{
  "stale": false,
  "type": "stack",
  "key": "stack-d_Rvaheu02xudocRnM-EZ",
  "index": 0,
  "routeNames": [
    "MainDrawer",
    "Login",
    "ProductQuickView",
    "HasuraPromoQuickView",
    "PromoQuickView",
    "ArticleShopProducts",
    "DeeplinkLoader",
    "AdoreSocietyModalScreen",
    "CMS",
    "NoInternetConnection"
  ],
  "routes": [
    {
      "key": "MainDrawer-qcdvNKHEDZ7OTgPbjkAI9",
      "name": "MainDrawer",
      "state": {
        "stale": false,
        "type": "drawer",
        "key": "drawer-ttg2qAkDtUciWJJAa7Kq8",
        "index": 0,
        "routeNames": [
          "Main"
        ],
        "history": [
          {
            "type": "route",
            "key": "Main-uBG3x92W3K7h3b-d819pm"
          }
        ],
        "routes": [
          {
            "name": "Main",
            "key": "Main-uBG3x92W3K7h3b-d819pm",
            "state": {
              "stale": false,
              "type": "stack",
              "key": "stack-aYsvtsb8JMnXdwECagzt9",
              "index": 1,
              "routeNames": [
                "MainTab",
                "Cart",
                "ProductStack",
                "GiftCertificate",
                "PostScreen",
                "BeautyIQPodcastEpisode",
                "OrderConfirm"
              ],
              "routes": [
                {
                  "key": "MainTab-BlVP17sWVFtGPi-QS-5KX",
                  "name": "MainTab",
                  "params": {
                    "initial": true,
                    "screen": "Shop",
                    "params": {
                      "initial": true,
                      "screen": "ShopCategoryProducts",
                      "params": {
                        "fromScreenPath": "/MainDrawer/Main/ProductStack?initial=true&params=%5Bobject%20Object%5D&screen=Product",
                        "showLoader": true,
                        "isDeepLink": true,
                        "url": "make-up",
                        "name": ""
                      }
                    }
                  },
                  "state": {
                    "stale": false,
                    "type": "tab",
                    "key": "tab-b2XMG1psmaTyt_QJunbV7",
                    "index": 1,
                    "routeNames": [
                      "Home",
                      "Shop",
                      "CartTab",
                      "Account",
                      "BeautyIQ"
                    ],
                    "history": [
                      {
                        "type": "route",
                        "key": "Home-hk8ll7oNJQwAU99t7mM8I"
                      },
                      {
                        "type": "route",
                        "key": "Shop-NTmuZzGOswQy_Qu7lWYBV"
                      }
                    ],
                    "routes": [
                      {
                        "name": "Home",
                        "key": "Home-hk8ll7oNJQwAU99t7mM8I",
                        "state": {
                          "stale": false,
                          "type": "stack",
                          "key": "stack-mvSosZ6jha7Kb66oOFvr5",
                          "index": 0,
                          "routeNames": [
                            "Home"
                          ],
                          "routes": [
                            {
                              "key": "Home-RxPeC6vASDcCRo3rJOh3H",
                              "name": "Home",
                              "params": {
                                "screenHeaderLogoSource": "https://images.ctfassets.net/pi28xy1s107o/52yUOD2lVWND8WFiTSi0ZL/140510078dd7784734f219dd16fee9be/ab-app-logo-christmas-rust-colour.png"
                              }
                            }
                          ]
                        }
                      },
                      {
                        "name": "Shop",
                        "key": "Shop-NTmuZzGOswQy_Qu7lWYBV",
                        "params": {
                          "initial": true,
                          "screen": "ShopCategoryProducts",
                          "params": {
                            "fromScreenPath": "/MainDrawer/Main/ProductStack?initial=true&params=%5Bobject%20Object%5D&screen=Product",
                            "showLoader": true,
                            "isDeepLink": true,
                            "url": "make-up",
                            "name": ""
                          },
                          "fromScreenPath": "/MainDrawer/Main/MainTab/Shop/ShopCategoryProducts?fromScreenPath=%2FMainDrawer%2FMain%2FMainTab%2FHome%2FHome%3FscreenHeaderLogoSource%3Dhttps%253A%252F%252Fimages.ctfassets.net%252Fpi28xy1s107o%252F52yUOD2lVWND8WFiTSi0ZL%252F140510078dd7784734f219dd16fee9be%252Fab-app-logo-christmas-rust-colour.png&isDeepLink=true&name=Night%20Cream&showLoader=true&url=skin-care%2Fmoisturisers%2Fnight",
                          "showLoader": true,
                          "isDeepLink": true
                        },
                        "state": {
                          "stale": false,
                          "type": "stack",
                          "key": "stack--YFUTNEnB8hmxLubCqokx",
                          "index": 1,
                          "routeNames": [
                            "Shop",
                            "ShopCategoryProducts",
                            "ShopTabs",
                            "ShopPromotions"
                          ],
                          "routes": [
                            {
                              "key": "Shop-EzEUbC3oDCUIXSTHhMS4s",
                              "name": "Shop",
                              "params": {
                                "fromScreenPath": "/MainDrawer/Main/MainTab/Shop/ShopCategoryProducts?fromScreenPath=%2FMainDrawer%2FMain%2FMainTab%2FHome%2FHome%3FscreenHeaderLogoSource%3Dhttps%253A%252F%252Fimages.ctfassets.net%252Fpi28xy1s107o%252F52yUOD2lVWND8WFiTSi0ZL%252F140510078dd7784734f219dd16fee9be%252Fab-app-logo-christmas-rust-colour.png&isDeepLink=true&name=Night%20Cream&showLoader=true&url=skin-care%2Fmoisturisers%2Fnight",
                                "showLoader": true,
                                "isDeepLink": true
                              }
                            },
                            {
                              "key": "ShopCategoryProducts-y7o-kdbSSlevCDIt5N-wh",
                              "name": "ShopCategoryProducts",
                              "params": {
                                "fromScreenPath": "/MainDrawer/Main/ProductStack?initial=true&params=%5Bobject%20Object%5D&screen=Product",
                                "showLoader": true,
                                "isDeepLink": true,
                                "url": "make-up",
                                "name": "Makeup"
                              }
                            }
                          ]
                        }
                      },
                      {
                        "name": "CartTab",
                        "key": "CartTab-zRnK01ka6OjAYp3IfYjIe"
                      },
                      {
                        "name": "Account",
                        "key": "Account-HgaUqlSE3VpFuxJ6KqCDd"
                      },
                      {
                        "name": "BeautyIQ",
                        "key": "BeautyIQ-3QJ3M1_VaGFwfF5VbhNNX"
                      }
                    ]
                  }
                },
                {
                  "key": "ProductStack-yjKOvF2B0sb4BZ827njFD",
                  "name": "ProductStack",
                  "params": {
                    "initial": true,
                    "screen": "Product",
                    "params": {
                      "fromScreenPath": "/MainDrawer/Main/MainTab/Shop/ShopCategoryProducts?fromScreenPath=%2FMainDrawer%2FMain%2FProductStack%3Finitial%3Dtrue%26params%3D%255Bobject%2520Object%255D%26screen%3DProduct&isDeepLink=true&name=Makeup&showLoader=true&url=make-up",
                      "showLoader": true,
                      "isDeepLink": true,
                      "identifier": "the-ordinary-hyaluronic-acid-2-b5-60ml"
                    }
                  }
                }
              ]
            }
          }
        ]
      }
    }
  ]
}
