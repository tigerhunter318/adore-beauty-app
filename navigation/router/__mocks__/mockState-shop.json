{
  "stale": false,
  "type": "stack",
  "key": "stack-ND4CpSjpTnb-1C3h4gs2h",
  "index": 0,
  "routeNames": [
    "MainDrawer",
    "Login",
    "ProductQuickView",
    "HasuraPromoQuickView",
    "PromoQuickView",
    "ArticleShopProducts",
    "DeeplinkLoader",
    "AdoreSocietyModalScreen",
    "CMS",
    "NoInternetConnection"
  ],
  "routes": [
    {
      "key": "MainDrawer-34orCYFcY9niQFI_evSQS",
      "name": "MainDrawer",
      "state": {
        "stale": false,
        "type": "drawer",
        "key": "drawer-cJnUbTewQUImrQEnIBwQI",
        "index": 0,
        "routeNames": [
          "Main"
        ],
        "history": [
          {
            "type": "route",
            "key": "Main-FkOPQVKNv1TVWc4cY_TKX"
          }
        ],
        "routes": [
          {
            "name": "Main",
            "key": "Main-FkOPQVKNv1TVWc4cY_TKX",
            "state": {
              "stale": false,
              "type": "stack",
              "key": "stack-JbYzuh7ntGhF8JvF4OMTj",
              "index": 0,
              "routeNames": [
                "MainTab",
                "Cart",
                "ProductStack",
                "GiftCertificate",
                "PostScreen",
                "BeautyIQPodcastEpisode",
                "OrderConfirm"
              ],
              "routes": [
                {
                  "key": "MainTab-odFB8UsmP9gftXngI-XiC",
                  "name": "MainTab",
                  "params": {
                    "initial": true,
                    "screen": "Home",
                    "params": {
                      "screenHeaderLogoSource": "",
                      "showLoader": true,
                      "isDeepLink": true
                    }
                  },
                  "state": {
                    "stale": false,
                    "type": "tab",
                    "key": "tab-WJuXdwSZG8qRyx_UUdfX_",
                    "index": 1,
                    "routeNames": [
                      "Home",
                      "Shop",
                      "CartTab",
                      "Account",
                      "BeautyIQ"
                    ],
                    "history": [
                      {
                        "type": "route",
                        "key": "Home-vV8xfWV_1GqXAc22EP4f4"
                      },
                      {
                        "type": "route",
                        "key": "Shop-JhjYbGnGKxk8M-5rYsUmM"
                      }
                    ],
                    "routes": [
                      {
                        "name": "Home",
                        "key": "Home-vV8xfWV_1GqXAc22EP4f4",
                        "params": {
                          "screenHeaderLogoSource": "",
                          "showLoader": true,
                          "isDeepLink": true
                        },
                        "state": {
                          "stale": false,
                          "type": "stack",
                          "key": "stack-gBYCx8kB6ErIdao2VoPVt",
                          "index": 0,
                          "routeNames": [
                            "Home"
                          ],
                          "routes": [
                            {
                              "key": "Home-ROu2hL5yOT18zEiciSo59",
                              "name": "Home",
                              "params": {
                                "screenHeaderLogoSource": "https://images.ctfassets.net/pi28xy1s107o/52yUOD2lVWND8WFiTSi0ZL/140510078dd7784734f219dd16fee9be/ab-app-logo-christmas-rust-colour.png"
                              }
                            }
                          ]
                        }
                      },
                      {
                        "name": "Shop",
                        "key": "Shop-JhjYbGnGKxk8M-5rYsUmM"
                      },
                      {
                        "name": "CartTab",
                        "key": "CartTab-D60pLbtMgi0T78KKbT9uV"
                      },
                      {
                        "name": "Account",
                        "key": "Account-S2fhVqdf3aULrZ-affjXm"
                      },
                      {
                        "name": "BeautyIQ",
                        "key": "BeautyIQ-nMTJBYgk9GX0ax737nRpu"
                      }
                    ]
                  }
                }
              ]
            }
          }
        ]
      }
    }
  ]
}
