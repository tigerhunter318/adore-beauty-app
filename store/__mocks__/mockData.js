import envConfig from '../../config/envConfig'

const testPassword = () => 'Passw0rd!'

const mockData = {
  account: {
    email: 'phil@test.com'
  },
  login: {
    email: envConfig.e2e.email,
    device_name: 'test',
    password: envConfig.e2e.password,
    first_name: 'App-Automated',
    last_name: 'Testing'
  },
  signup: {
    email: '@example.com',
    first_name: 'phil',
    last_name: 'test',
    password: testPassword(),
    password_confirmation: testPassword(),
    device_name: 'test',
    au_store: 1,
    nz_store: 0
  },
  addToCartMultiple: [
    { productSku: 'E9WK010000' },
    { productSku: 'abmbdbag' },
    { productSku: 'SC', cartProductId: '29347' }
  ],
  addToCart: {
    // productSku: 'SKSC20019',
    productSku: '4979006053241', // staging only
    quantity: 1,
    product_id: '',
    cartProductId: 42886
  },
  addAddress: {
    firstName: 'Phil',
    lastName: 'Test',
    phone: '0404 000 000',
    addressMeta: {
      address_line_1: '44A Broadway',
      address_line_2: null,
      locality_name: 'BONBEACH',
      state_territory: 'VIC',
      postcode: '3196'
    }
  },
  billingAddress: {
    first_name: 'Phil',
    last_name: 'Test',
    email: 'phil@example.com',
    address1: '44A Broadway',
    address2: null,
    city: 'BONBEACH',
    state_or_province: 'VIC',
    state_or_province_code: '',
    postal_code: '3196',
    phone: '0404539959',
    country: 'Australia',
    country_code: 'AU'
  },
  card: {
    type: 'card',
    number: '4111111111111111', // paymentDetails.cardNumber,
    cardholder_name: 'test card', // paymentDetails.cardName,
    expiry_month: 12,
    expiry_year: 2020,
    verification_value: '123' // paymentDetails.cardCsv
  }
}

export default mockData
