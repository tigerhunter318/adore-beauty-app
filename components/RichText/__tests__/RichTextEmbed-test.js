import React from 'react'
import renderer from 'react-test-renderer'
import RichTextEmbed from '../RichTextEmbed'
import { getArticleContent } from './__mocks__/helper'

it(`RichTextEmbed renders alterna-air-dry-balm-review instagram embed`, () => {
  const parsedContent = getArticleContent(3, 'alterna-air-dry-balm-review')
  const tree = renderer.create(<RichTextEmbed content={parsedContent} />).toJSON()

  expect(tree).toMatchSnapshot()
})

it(`RichTextEmbed renders what-is-a-serum instagram embed`, () => {
  const parsedContent = getArticleContent(7, 'what-is-a-serum')
  const tree = renderer.create(<RichTextEmbed content={parsedContent} />).toJSON()

  expect(tree).toMatchSnapshot()
})
