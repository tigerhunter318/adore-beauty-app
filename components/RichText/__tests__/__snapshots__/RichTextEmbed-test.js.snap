// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextEmbed renders alterna-air-dry-balm-review instagram embed 1`] = `
<View
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
          },
          Object {},
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <RNCWebView
            androidLayerType="hardware"
            cacheEnabled={true}
            domStorageEnabled={true}
            incognito={true}
            injectedJavaScript="
  function getHeight(el) {
      var elHeight = el.scrollHeight;
      var docHeight = document.body.scrollHeight;
      var height;
      if (elHeight < docHeight && elHeight > 0) {
          height = elHeight;
      } else {
          height = docHeight;
      }
      return height;
  }

  var height = 0;
  var wrapper = document.body.firstChild;
  function updateSize() {
      var h = getHeight(wrapper);
      if (h !== height) {
          height = h;
          window.ReactNativeWebView.postMessage(JSON.stringify({\\"height\\": height}));
      }
  }

  window.addEventListener(\\"message\\", function(e) {
      // Listen to all messages to trigger additional size checks.
      // Includes Iframely resize message, native pings from Twitter & et al
      
      // You can also try checking sizes periodically if you have issues
      // see https://github.com/react-native-community/react-native-webview/issues/154

      updateSize();
  });
  window.addEventListener(\\"load\\", function(e) {
    updateSize();
  })
  updateSize()
"
            injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
            injectedJavaScriptForMainFrameOnly={true}
            javaScriptEnabled={true}
            messagingEnabled={true}
            onContentProcessDidTerminate={[Function]}
            onHttpError={[Function]}
            onLoadingError={[Function]}
            onLoadingFinish={[Function]}
            onLoadingProgress={[Function]}
            onLoadingStart={[Function]}
            onMessage={[Function]}
            onShouldStartLoadWithRequest={[Function]}
            scrollEnabled={false}
            source={
              Object {
                "html": "<head>
<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, viewport-fit=cover\\" />
<style>
.iframely-responsive {
  top: 0; left: 0; width: 100%; height: 0;
  position: relative; padding-bottom: 56.25%;
}
.iframely-responsive>* {
  top: 0; left: 0; width: 100%; height: 100%; position: absolute; border: 0;
}
</style>
<script async src=\\"https://www.instagram.com/embed.js\\"></script><script src=\\"https://cdn.embedly.com/widgets/platform.js\\"></script><script>window.instgrm.Embeds.process()</script></head><body><blockquote class=\\"instagram-media\\" data-instgrm-permalink=\\"https://www.instagram.com/p/CX9CYy1hAc8/?utm_source=ig_embed&utm_campaign=loading\\" data-instgrm-version=\\"14\\" style=\\" background:#FFF; border:0; border-radius:3px; box-shadow:0 0 1px 0 rgba(0,0,0,0.5),0 1px 10px 0 rgba(0,0,0,0.15); margin: 1px; max-width:540px; min-width:326px; padding:0; width:99.375%; width:-webkit-calc(100% - 2px); width:calc(100% - 2px);; max-width:100% !important; min-width:auto !important;\\"><div style=\\"padding:16px;\\"> <a href=\\"https://www.instagram.com/p/CX9CYy1hAc8/?utm_source=ig_embed&utm_campaign=loading\\" style=\\" background:#FFFFFF; line-height:0; padding:0 0; text-align:center; text-decoration:none; width:100%;\\" target=\\"_blank\\" data-card-key=\\"542239e8d5144167ac6474a184f2d244\\"> <div style=\\" display: flex; flex-direction: row; align-items: center;\\"> <div style=\\"display: flex; flex-direction: column; flex-grow: 1; justify-content: center;\\"> </div></div> <div style=\\"display:block; height:50px; margin:0 auto 12px; width:50px;\\"><svg width=\\"50px\\" height=\\"50px\\" viewBox=\\"0 0 60 60\\" version=\\"1.1\\" xmlns=\\"https://www.w3.org/2000/svg\\" xmlns:xlink=\\"https://www.w3.org/1999/xlink\\"><g stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><g transform=\\"translate(-511.000000, -20.000000)\\" fill=\\"#000000\\"><g/></g></g></svg></div><div style=\\"padding-top: 8px;\\"> <div style=\\" color:#3897f0; font-family:Arial,sans-serif; font-size:14px; font-style:normal; font-weight:550; line-height:18px;\\">View this post on Instagram</div></div> <div style=\\"display: flex; flex-direction: row; margin-bottom: 14px; align-items: center;\\"><div> </div><div style=\\"margin-left: 8px;\\"> </div><div style=\\"margin-left: auto;\\"> </div></div> <div style=\\"display: flex; flex-direction: column; flex-grow: 1; justify-content: center; margin-bottom: 24px;\\"> </div></a><p style=\\" color:#c9c8cd; font-family:Arial,sans-serif; font-size:14px; line-height:17px; margin-bottom:0; margin-top:8px; overflow:hidden; padding:8px 0 7px; text-align:center; text-overflow:ellipsis; white-space:nowrap;\\"><a href=\\"https://www.instagram.com/p/CX9CYy1hAc8/?utm_source=ig_embed&utm_campaign=loading\\" style=\\" color:#c9c8cd; font-family:Arial,sans-serif; font-size:14px; font-style:normal; font-weight:normal; line-height:17px; text-decoration:none;\\" target=\\"_blank\\" data-card-key=\\"542239e8d5144167ac6474a184f2d244\\">A post shared by Alterna Haircare (@alternahaircare)</a></p></div></blockquote></body>",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "height": 0,
                },
              ]
            }
            textInteractionEnabled={true}
            useSharedProcessPool={true}
            useWebKit={true}
          />
          <ActivityIndicator
            animating={false}
            color="#ACACAC"
            size="small"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`RichTextEmbed renders what-is-a-serum instagram embed 1`] = `
<View
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
          },
          Object {},
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <RNCWebView
            androidLayerType="hardware"
            cacheEnabled={true}
            domStorageEnabled={true}
            incognito={true}
            injectedJavaScript="
  function getHeight(el) {
      var elHeight = el.scrollHeight;
      var docHeight = document.body.scrollHeight;
      var height;
      if (elHeight < docHeight && elHeight > 0) {
          height = elHeight;
      } else {
          height = docHeight;
      }
      return height;
  }

  var height = 0;
  var wrapper = document.body.firstChild;
  function updateSize() {
      var h = getHeight(wrapper);
      if (h !== height) {
          height = h;
          window.ReactNativeWebView.postMessage(JSON.stringify({\\"height\\": height}));
      }
  }

  window.addEventListener(\\"message\\", function(e) {
      // Listen to all messages to trigger additional size checks.
      // Includes Iframely resize message, native pings from Twitter & et al
      
      // You can also try checking sizes periodically if you have issues
      // see https://github.com/react-native-community/react-native-webview/issues/154

      updateSize();
  });
  window.addEventListener(\\"load\\", function(e) {
    updateSize();
  })
  updateSize()
"
            injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
            injectedJavaScriptForMainFrameOnly={true}
            javaScriptEnabled={true}
            messagingEnabled={true}
            onContentProcessDidTerminate={[Function]}
            onHttpError={[Function]}
            onLoadingError={[Function]}
            onLoadingFinish={[Function]}
            onLoadingProgress={[Function]}
            onLoadingStart={[Function]}
            onMessage={[Function]}
            onShouldStartLoadWithRequest={[Function]}
            scrollEnabled={false}
            source={
              Object {
                "html": "<head>
<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, viewport-fit=cover\\" />
<style>
.iframely-responsive {
  top: 0; left: 0; width: 100%; height: 0;
  position: relative; padding-bottom: 56.25%;
}
.iframely-responsive>* {
  top: 0; left: 0; width: 100%; height: 100%; position: absolute; border: 0;
}
</style>
<script async src=\\"https://www.instagram.com/embed.js\\"></script><script src=\\"https://cdn.embedly.com/widgets/platform.js\\"></script><script>window.instgrm.Embeds.process()</script></head><body><blockquote class=\\"instagram-media\\" data-instgrm-permalink=\\"https://www.instagram.com/p/CVpehWZhCYU/?utm_source=ig_embed&utm_campaign=loading\\" data-instgrm-version=\\"14\\" style=\\" background:#FFF; border:0; border-radius:3px; box-shadow:0 0 1px 0 rgba(0,0,0,0.5),0 1px 10px 0 rgba(0,0,0,0.15); margin: 1px; max-width:540px; min-width:326px; padding:0; width:99.375%; width:-webkit-calc(100% - 2px); width:calc(100% - 2px);; max-width:100% !important; min-width:auto !important;\\"><div style=\\"padding:16px;\\"> <a href=\\"https://www.instagram.com/p/CVpehWZhCYU/?utm_source=ig_embed&utm_campaign=loading\\" style=\\" background:#FFFFFF; line-height:0; padding:0 0; text-align:center; text-decoration:none; width:100%;\\" target=\\"_blank\\" data-card-key=\\"542239e8d5144167ac6474a184f2d244\\"> <div style=\\" display: flex; flex-direction: row; align-items: center;\\"> <div style=\\"display: flex; flex-direction: column; flex-grow: 1; justify-content: center;\\"> </div></div> <div style=\\"display:block; height:50px; margin:0 auto 12px; width:50px;\\"><svg width=\\"50px\\" height=\\"50px\\" viewBox=\\"0 0 60 60\\" version=\\"1.1\\" xmlns=\\"https://www.w3.org/2000/svg\\" xmlns:xlink=\\"https://www.w3.org/1999/xlink\\"><g stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><g transform=\\"translate(-511.000000, -20.000000)\\" fill=\\"#000000\\"><g/></g></g></svg></div><div style=\\"padding-top: 8px;\\"> <div style=\\" color:#3897f0; font-family:Arial,sans-serif; font-size:14px; font-style:normal; font-weight:550; line-height:18px;\\">View this post on Instagram</div></div> <div style=\\"display: flex; flex-direction: row; margin-bottom: 14px; align-items: center;\\"><div> </div><div style=\\"margin-left: 8px;\\"> </div><div style=\\"margin-left: auto;\\"> </div></div> <div style=\\"display: flex; flex-direction: column; flex-grow: 1; justify-content: center; margin-bottom: 24px;\\"> </div></a><p style=\\" color:#c9c8cd; font-family:Arial,sans-serif; font-size:14px; line-height:17px; margin-bottom:0; margin-top:8px; overflow:hidden; padding:8px 0 7px; text-align:center; text-overflow:ellipsis; white-space:nowrap;\\"><a href=\\"https://www.instagram.com/p/CVpehWZhCYU/?utm_source=ig_embed&utm_campaign=loading\\" style=\\" color:#c9c8cd; font-family:Arial,sans-serif; font-size:14px; font-style:normal; font-weight:normal; line-height:17px; text-decoration:none;\\" target=\\"_blank\\" data-card-key=\\"542239e8d5144167ac6474a184f2d244\\">A post shared by BIODERMA Australia (@biodermaaustralia)</a></p></div></blockquote></body>",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "height": 0,
                },
              ]
            }
            textInteractionEnabled={true}
            useSharedProcessPool={true}
            useWebKit={true}
          />
          <ActivityIndicator
            animating={false}
            color="#ACACAC"
            size="small"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
